### PR TITLE
copr: fix `tail_len` handle in `process_new_collation_kv`

### PR DESCRIPTION
**Pull request by iosmanthus**
Signed-off-by: iosmanthus <myosmanthustree@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/19996 

Problem Summary:

Missing index value's `tail_len` handling in `process_new_collation_kv`;

What's Changed:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Fix `tail_len` handle in `process_new_collation_kv`

### CODE REVIEW

File: components/tidb_query_vec_executors/src/index_scan_executor.rs [hunk 1]
```diff
@@ -359,6 +359,7 @@ impl IndexScanExecutorImpl {
         value: &[u8],
         columns: &mut LazyBatchColumnVec,
     ) -> Result<()> {
+        let tail_len = value[0] as usize;
```
Comment (zhongzc) [inline components/tidb_query_vec_executors/src/index_scan_executor.rs:362]: The comment above says `tail_len` is always `0x00`. In what cases does the rule break?

Comment (coocood) [reply components/tidb_query_vec_executors/src/index_scan_executor.rs:362]: In the case tidb mistakenly write untouched key to tikv.

Comment (zhongzc) [reply components/tidb_query_vec_executors/src/index_scan_executor.rs:362]: @iosmanthus Could you update comments for that case?

Comment (iosmanthus) [reply components/tidb_query_vec_executors/src/index_scan_executor.rs:362]: OK

**Comment by coocood** [comment]
It's fixed by https://github.com/pingcap/tidb/pull/20063

**Comment by Xuanwo** [comment]
So we don't need to fix it in TiKV side?

**Comment by breeswish** [comment]
Do we still need this PR?

**Pull request closed without merging**